packages/server/src/whip/whipEndpoint.ts

```diff
@@ -1,150 +1,181 @@
+    this.interfaceIp = opts?.interfaceIp || "0.0.0.0";
+    this.useHttps = !!(opts?.https);
+    this.hostname = opts?.hostname;
+    this.enabledWrtcPlugins = opts?.enabledWrtcPlugins || [];
+    this.iceServers = opts?.iceServers || [];
+    this.tls = opts?.tls;
+    this.resourceManagerUrl = opts?.resourceManagerUrl
+    this.territoryCode = opts?.territoryCode
+
+    let httpsServer;
+    if (this.useHttps && this.tls) {
+      httpsServer = https.createServer({ key: this.tls.key, cert: this.tls.cert });
+    }    
+    this.server = fastify({ 
+      ignoreTrailingSlash: true, 
+      logger: { level: "info" },
+      https: httpsServer,
+    });
+    this.server.register(require("fastify-cors"), {
+      exposedHeaders: ["Location", "ETag", "Link", "Access-Control-Allow-Methods"],
+      methods: ["POST", "GET", "OPTIONS", "DELETE", "PATCH"],
+      preflightContinue: true,
+      strictPreflight: false,
+    });
+    this.server.register(api, { prefix: "/api/v2", instance: this });
+    this.server.get("/", async () => {
+      return "OK\n";
+    });
+    this.resources = {};
+  }
+
+  registerBroadcasterClient(broadcasterClientSfuPair: BroadcasterClientSfuPair) {
+    this.broadcasterClientSfuPairs.push(broadcasterClientSfuPair);
+  }
+
+  setOriginSfuUrl(url: string) {
+    this.originSfuUrl = url;
+  }
+
+  setSfuApiKey(apiKey: string) {
+    this.sfuApiKey = apiKey;
+  }
+
+  hasBroadcasterClient(): boolean {
+    return this.broadcasterClientSfuPairs.length !== 0;
+  }
+
+  getBroadcasterClientSfuPairs(): BroadcasterClientSfuPair[] {
+    return this.broadcasterClientSfuPairs;
+  }
+
+  getOriginSfuUrl(): string {
+    return this.originSfuUrl;
+  }
+
+  getSfuApiKey(): string {
+    return this.sfuApiKey;
+  }
+
+  addResource(resource: WhipResource) {
+    this.resources[resource.getId()] = resource;
+  }
+
+  deleteResource(id: string) {
+    const resource = this.resources[id]; 
+    if (resource) {
+      delete this.resources[id];
+      resource.destroy();
+    }
+  }
+
+  async patchResource(id: string, body: string, eTag:string | undefined): Promise<number> {
+    const resource = this.resources[id]; 
+    if (resource) {
+      return resource.patch(body, eTag);
+    }
+    return 404;
+  }
+
+  listResources() {
+    return Object.keys(this.resources).map(id => {
+      const resource = this.resources[id];
+      return { id: resource.getId(), type: resource.getType() };
+    });
+  }
+
+  getEnabledPlugins() {
+    return this.enabledWrtcPlugins;
+  }
+
+  getResourceById(resourceId: string) {
+    return this.resources[resourceId];
+  }
+
+  getIceServers(): WhipResourceIceServer[]|null {
+    return this.iceServers;
+  }
+
+  getServerAddress(): string | undefined {
+    if (this.hostname) {
+      return (this.useHttps ? "https" : "http") + "://" + this.hostname + ":" + this.extPort;
+    }
+    return undefined;
+  }
+
+  readEdgeListFromFile(): OriginsAndEdgesPerTerritory {

```

Comment:
I suggest to return a promise (async) here as well so it aligns with the `readEdgeListFromService()` method